### DISCUSSION TITLE
Config file issue

### DISCUSSION THREAD

**Discussion by uSlackr** [gboudreau/Greyhole#254] [Q&A] [answered]
I upgraded to the latest 15.x version from the repo in Debian yesterday. Greyhole would not start.  Here's the error:
Starting Greyhole ... Dec 22 01:07:59 ERROR test-config: You have no 'storage_pool_drive'
CRITICAL  test-config: Config file parsing failed

It looks as though the up[grade package renamed the current config and replaced it with a generic one.
original was renamed "greyhole.conf.dpkg-old".  Is this a "bug" ?

**Comment by gboudreau** [comment, marked as answer]
The `/etc/greyhole.conf` file is listed in the .deb `conffiles` list, meaning that `dpkg` will monitor both user modifications to this file, and changes in the packaged version of this file. If the user edited the file (your case), then installing a new package that contains a modified `greyhole.conf` file will prompt you what should be done with this file with something that will look like this:

```
Configuration file `/etc/greyhole.conf'
 ==> Modified (by you or by a script) since installation.
 ==> Package distributor has shipped an updated version.
   What would you like to do about it ?  Your options are:
    Y or I  : install the package maintainer's version
    N or O  : keep your currently-installed version
      D     : show the differences between the versions
      Z     : start a shell to examine the situation
 The default action is to keep your current version.
*** greyhole.conf (Y/I/N/O/D/Z) [default=N] ? 
```


You can read more details on dpkg handling of `conffiles` here : https://raphaelhertzog.com/2010/09/21/debian-conffile-configuration-file-managed-by-dpkg/

I would guess you might have answered `Y/I` inadvertently, or you used a `dpkg`/`apt` command to upgrade that defaulted to installing the new version over yours.
